### PR TITLE
add EKP learning rate schedulers 

### PR DESCRIPTION
**Pull request by costachris**
## Changes

1. Adds functionality for using native learning rate schedulers from EKP.jl
2. Outputs timestep history to `Diagnostics.nc` output file.
3. Make EKP verbose

### CODE REVIEW

File: src/KalmanProcessUtils.jl [hunk 1]
```diff
@@ -244,5 +248,37 @@ end
...
+end
+
+function modify_field(scheduler::T, field_name::Symbol, new_value) where T
+    fields = fieldnames(T)
+    values = [field_name == f ? new_value : getfield(scheduler, f) for f in fields]
+    return T(values...)
+end
+
+"""Return new ekp object with updated scheduler object."""
+function update_ek_timestepper(ekp, iteration)
```
Comment (costachris) [inline src/KalmanProcessUtils.jl:264]: This is needed to correctly update the EKP scheduler in calibrations with mini-batches. By default, `calculate_timestep!` only updates the `scheduler.inv_sqrt_noise` field on the first step (when `ekp.Δt` is empty). With mini-batching the obs_noise_cov changes in each iteration, so we need to regenerate the scheduler object.

**Review by odunbar — APPROVED** [review]
LGTM - my comments are only questions... 
v1.1.4 release is coming soon - I need to satisfy some new compat requirements before it will work though

File: src/Pipeline.jl [hunk 2]
```diff
@@ -59,6 +59,12 @@ function init_calibration(config::Dict{Any, Any}; mode::String = "hpc", job_id::
     Δt_scheduler = get_entry(proc_config, "Δt", 1.0)
     Δt = get_Δt(Δt_scheduler, 1)
 
+    scheduler = get_entry(proc_config, "scheduler", nothing)
+
+    if !isnothing(scheduler)
```
[the remark this answers is not in the document]
Comment (costachris) [reply src/Pipeline.jl:64]: Yes, there are also CEDMF learning rate schedulers (Piecewise growth/decay), and they work by setting $\Delta t $ directly. They will probably be deprecated soon, but this is just to ensure backwards compatibility for now.

**Pull request merged**